components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -61,6 +61,30 @@ pub fn from_days(ctx: &mut EvalContext, arg: &Option<Int>) -> Result<Option<Time
     })
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn date_diff(
+    ctx: &mut EvalContext,
+    arg1: &Option<Time>,
+    arg2: &Option<Time>,
+) -> Result<Option<Int>> {
+    if let (Some(lhs), Some(rhs)) = (arg1, arg2) {
+        if lhs.invalid_zero() {
+            return ctx
+                .handle_invalid_time_error(Error::incorrect_datetime_value(&format!("{}", lhs)))
```

Comment:
`Error::incorrect_datetime_value(lhs)` is enough.

Reply:
Got it, please check  https://github.com/tikv/tikv/pull/6043/commits/2aaa0e844970ce558f325303b716583a9bf49285